tiledb/sm/filesystem/gcs.cc

```diff
@@ -551,6 +552,83 @@ tuple<Status, optional<std::vector<directory_entry>>> GCS::ls_with_sizes(
   return {Status::Ok(), entries};
 }
 
+LsObjects GCS::ls_filtered_impl(
+    const URI& uri,
+    std::function<bool(const std::string_view, uint64_t)> file_filter,
+    bool recursive) const {
+  throw_if_not_ok(init_client());
+
+  const URI uri_dir = uri.add_trailing_slash();
+
+  if (!uri_dir.is_gcs()) {
+    throw StatusException(Status_GCSError(
+        std::string("URI is not a GCS URI: " + uri_dir.to_string())));
+  }
+
+  std::string bucket_name;
+  std::string object_path;
+  bool is_gcs = stdx::string::starts_with(uri_dir.to_string(), "gcs://");
```

Comment:
The `bool` is fine, just a suggestion if you want to remove inline ternary operator checks while assembling full URIs.

```c++
std::string prefix = uri_dir.backend_name() + "://"
```

Reply:
Didn't know about it, thanks!